### PR TITLE
HPCC-16525 LDAP Connection reconnects without SSL

### PR DESCRIPTION
**Pull request by RussWhitehead**
If secmgr is configured to use LDAPS, it initially connects using the secure
port. But when the connection goes stale the reconnect always uses the
unsecure port 383. This PR remembers the selected protocol and reuses it
on reconnect

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@rpastrana Please review

**Comment by richardkchapman** [comment]
Should this target 6.0.8?

**Comment by RussWhitehead** [comment]
I think 6.0.8 would be better if we can get it in. However, nobody reported it, I just noticed it when I was looking at that class

File: system/security/LdapSecurity/ldapconnection.cpp [hunk 1]
```diff
@@ -688,6 +690,7 @@ class CLdapConnection : implements ILdapConnection, public CInterface
         }
     }
 
+private:
     virtual int connect(const char* ldapserver, const char* protocol)
```
Comment (rpastrana) [inline system/security/LdapSecurity/ldapconnection.cpp]: Is it possible this version of connect is being used? (is it possible a plugin which resides in a separate repo is currently referencing this method)? Most likely ok, but making sure we always consider when changing access rules.

**Review by rpastrana — APPROVED** [review]
Minor comment, otherwise looks good.

File: system/security/LdapSecurity/ldapconnection.cpp (hunk 1 above)
Comment (RussWhitehead) [inline system/security/LdapSecurity/ldapconnection.cpp]: I built with LN branch successfully.  But you do make a good general point about changing the interface on any externally facing objects. Good topic to discuss in another Jira?

**Comment by RussWhitehead** [comment]
@richardkchapman Please merge and close

**Pull request merged**